compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1490,7 +1490,7 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                             format!(
                                 "{}{}{} {}{}",
                                 if sp.is_desugaring(DesugaringKind::Async) {
-                                    "the `Output` of this `async fn`'s "
+                                    "checked the return type of this `async fn`, "
```

Comment:
This doesn't look any more clear to me. Maybe add ``.note("while checking the return type of this `async fn`")`` instead?

Reply:
That does make sense! I'll get that incorporated in.

Reply:
Done! Moved the changes to #82165